library/Network/IPFS/Pin.hs

```diff
@@ -46,21 +57,49 @@ rm cid = ipfsUnpin cid False >>= \case
         logDebug <| "Pinned CID " <> display cid'
         return <| Right cid'
 
-      _ ->
-        logLeft <| UnexpectedOutput <| UTF8.textShow cids
+      _ -> do
+        formattedErr <- parseUnexpectedOutput <| UTF8.textShow cids
+        return <| Left <| formattedErr
 
   Left _ -> do
     logDebug <| "Cannot unpin CID " <> display cid <> " because it was not pinned"
     return <| Right cid
 
-logLeft ::
+-- | Parse and Log the Servant Client Error returned from the IPFS Daemon
+parseClientError ::
+  ( MonadRIO        cfg m
+  , HasLogFunc      cfg
+  )
+  => ClientError
+  -> m (Error)
+parseClientError err = do
+  logError <| displayShow err
+  let newError = case err of
+        (FailureResponse _ response) -> do
+          response
+          |> responseBody
+          |> decode
+          |> \case
```

Comment:
For consistency, we should have pipes indented under `response`. This indentation will confuse the parser in some contexts.

Reply:
...also do you need the `do` here?

Reply:
Also, no need for those wrapping parens:

```haskell
(FailureResponse _ response) -> do
FailureResponse _ response -> do
```